key.go

```diff
@@ -0,0 +1,39 @@
+package dataloader
+
+// Key is the interface that all keys need to implement
+type Key interface {
+	// Key returns a guaranteed unique string that can be used to identify an object
+	Key() string
```

Comment:
Now that the interface is named `Key`, the method `Key()` stutters. What do you think about renaming this method `String()`?

```go
type Key interface {
  String() string
  Raw()    interface{}
}
```

Reply:
yeah, that makes sense to me. I'll do it.